### PR TITLE
HW-51710 get user details in the core sdk

### PR DESCRIPTION
**Pull request by vshcherbyna-epam**

### CODE REVIEW

File: core/src/main/java/com/hyperwallet/android/Hyperwallet.java [hunk 1]
```diff
@@ -250,6 +249,30 @@ public void getBankCard(@NonNull final String transferMethodToken,
...
+     * processing the request.</p>
+     *
+     * <p>This function will requests a new authentication token via {@link HyperwalletAuthenticationTokenProvider}
+     * if the current one is expired or about to expire.</p>
+     *
+     * @param userToken the Hyperwallet specific unique identifier for the {@code HyperwalletUser}
+     *                  being requested; must not be null
+     * @param listener  the callback handler of responses from the Hyperwallet platform; must not be null
+     */
+    public void getUser(@NonNull String userToken,
```
Comment (azakrevska-epam) [inline core/src/main/java/com/hyperwallet/android/Hyperwallet.java]: @vshcherbyna-epam I think you don't need userToken here as param, could you remove it?

Comment (vshcherbyna-epam) [reply core/src/main/java/com/hyperwallet/android/Hyperwallet.java]: thank you, removed

File: core/src/main/java/com/hyperwallet/android/model/HyperwalletUser.java [hunk 2]
```diff
@@ -0,0 +1,567 @@
+/*
+ * Copyright 2018 Hyperwallet
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
```
Comment (fmattos-hw) [inline core/src/main/java/com/hyperwallet/android/model/HyperwalletUser.java]: Use the same agreement from the PayPal account

File: core/src/main/java/com/hyperwallet/android/model/HyperwalletUser.java [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+     * </p>
+     *
+     * @param key can only be a {@link String} that represents a {@link HyperwalletUser.UserField} name
+     * @return a {@link String} value that represents the value of a {@link HyperwalletField}
+     */
+    @Nullable
+    public String getField(@NonNull @UserField String key) {
+        return mFields.get(key) != null ? (String) mFields.get(key) : null;
+    }
+
```
Comment (fmattos-hw) [inline core/src/main/java/com/hyperwallet/android/model/HyperwalletUser.java]: please add getters to all those fields. The getters are just wrappers of the getFields

File: core/src/main/java/com/hyperwallet/android/model/HyperwalletUser.java [hunk 4]
```diff
@@ -0,0 +1,567 @@
...
+                public HyperwalletUser[] newArray(int size) {
+                    return new HyperwalletUser[0];
+                }
+            };
+
+    protected void setFields(@NonNull Map<String, Object> fields) {
+        mFields = fields;
+    }
+
+    public static class Builder {
```
Comment (fmattos-hw) [inline core/src/main/java/com/hyperwallet/android/model/HyperwalletUser.java]: Please add @Nullable and final to the method arguments

File: core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+                mExternalResourceManager.getResourceContent("user_response.json"),
+                new TypeReference<HyperwalletUser>() {
+                });
+
+        assertThat(actualUser, is(notNullValue()));
+        assertThat(actualUser.getField(TOKEN), is("usr-f9154016-94e8-4686-a840-075688ac07b5"));
+        assertThat(actualUser.getField(STATUS), is("PRE_ACTIVATED"));
+        assertThat(actualUser.getField(VERIFICATION_STATUS), is("NOT_REQUIRED"));
+        assertThat(actualUser.getField(CREATED_ON), is("2017-10-30T22:15:45"));
+        assertThat(actualUser.getField(CLIENT_USER_ID), is("CSK7b8Ffch"));
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java]: please use 123345789

File: core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+        assertThat(actualUser.getField(PROFILE_TYPE), is("INDIVIDUAL"));
+        assertThat(actualUser.getField(FIRST_NAME), is("Some"));
+        assertThat(actualUser.getField(LAST_NAME), is("Guy"));
+        assertThat(actualUser.getField(DATE_OF_BIRTH), is("1991-01-01"));
+        assertThat(actualUser.getField(EMAIL), is("user+4satF1xV@hyperwallet.com"));
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java]: use someguy@hyperwallet.com

File: core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+
+    @Test
+    public void testToJsonObject_user() throws Exception {
+
+        final HyperwalletUser expectedUser = new HyperwalletUser.Builder()
+                .token("usr-f9154016-94e8-4686-a840-075688ac07b5")
+                .status(LOCKED)
+                .verificationStatus(UNDER_REVIEW)
+                .createdOn("2017-10-30T22:15:45")
+                .clientUserId("CSK7b8Ffch")
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java]: same here 123456789

File: core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+                .addressLine1("575 Market Street")
+                .addressLine2("Valenka 4b")
+                .businessContactRole(OWNER)
+                .businessName("My Company")
+                .businessRegistrationCountry("Canada")
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java]: CA

File: core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+    public void testToJsonObject_user() throws Exception {
+
+        final HyperwalletUser expectedUser = new HyperwalletUser.Builder()
+                .token("usr-f9154016-94e8-4686-a840-075688ac07b5")
+                .status(LOCKED)
+                .verificationStatus(UNDER_REVIEW)
+                .createdOn("2017-10-30T22:15:45")
+                .clientUserId("CSK7b8Ffch")
+                .addressLine1("575 Market Street")
+                .addressLine2("Valenka 4b")
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java]: for address line 247 Tottenham Court Rd

File: core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+                .businessContactRole(OWNER)
+                .businessName("My Company")
+                .businessRegistrationCountry("Canada")
+                .businessRegistrationId("werqq")
+                .businessRegistrationStateProvince("Current")
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java]: BC

File: core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java [hunk 11]
```diff
@@ -0,0 +1,265 @@
...
+                .businessContactAddressLine1("34 Zoo pl")
+                .businessContactAddressLine2("102 Park ave")
+                .businessContactCity("Zippy")
+                .businessContactStateProvince("Quebec")
+                .businessContactCountry("Canada")
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java]: CA

File: core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java [hunk 12]
```diff
@@ -0,0 +1,265 @@
...
+                .businessContactPostalCode("QC H2Y 2E2")
+                .businessOperatingName("OP name")
+                .businessType(CORPORATION)
+                .city("Montréal")
+                .country("Canada")
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java]: CA

File: core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java [hunk 13]
```diff
@@ -0,0 +1,265 @@
...
+                .countryOfBirth("USA")
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java]: US

File: core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java [hunk 14]
```diff
@@ -0,0 +1,265 @@
...
+                .countryOfNationality("Canada")
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java]: CA

File: core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java [hunk 15]
```diff
@@ -0,0 +1,265 @@
...
+                .language("fr")
+                .lastName("Smith")
+                .middleName("de")
+                .mobileNumber("+1514-496-7678")
+                .passportId("AV54467")
+                .phoneNumber("496-7678")
+                .postalCode("94105")
+                .profileType(INDIVIDUAL)
+                .programToken("prg-83836cdf-2ce2-4696-8bc5-f1b86077238c")
+                .stateProvince("Old Port")
```
Comment (fmattos-hw) [inline core/src/test/java/com/hyperwallet/android/model/HyperwalletUserTest.java]: BC

File: core/src/main/java/com/hyperwallet/android/model/HyperwalletUser.java (hunk 3 above)
Comment (vshcherbyna-epam) [reply core/src/main/java/com/hyperwallet/android/model/HyperwalletUser.java]: added

**Pull request merged**